### PR TITLE
fio_linux: Add a case run fio for Linux

### PR DESCRIPTION
**Pull request by YongxueHong**
Boot guest with "aio=native" or "aio=threads" CLI option
and run fio tools.

ID: 1630819
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on https://github.com/autotest/tp-qemu/pull/1614

**Comment by YongxueHong** [comment]
@XueqiangWei @CongLi  @ngu-niny  @hereischen  help review, thanks.

File: qemu/tests/cfg/fio_linux.cfg [hunk 1]
```diff
@@ -0,0 +1,22 @@
+- fio_linux:
+    type = fio_linux
+    only Linux
+    only virtio_blk virtio_scsi
```
Comment (ngu-niny) [inline qemu/tests/cfg/fio_linux.cfg]: Please add spapr_vscsi, the test could PASS:
 (1/2) Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.fio_linux.aio_native: PASS (390.53 s)
 (2/2) Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.fio_linux.aio_threads: PASS (437.30 s)

File: qemu/tests/fio_linux.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    logging.info('Boot a guest with %s.' % ', '.join(info))
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=float(params.get("login_timeout", 240)))
+    try:
+        fio = generate_instance(params, session, 'fio')
+        for did in _get_data_disks():
+            for option in params['fio_options'].split(';'):
+                fio.run(' '.join(('--filename=%s' % did, option)))
```
Comment (hereischen) [inline qemu/tests/fio_linux.py]: why not use just string format？
```python
fio.run(''--filename=%s %s' % (did, option))
```

Comment (YongxueHong) [reply qemu/tests/fio_linux.py]: thanks, have updated.

File: qemu/tests/fio_linux.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            if match:
+                drive_id = match.group(2)
+            else:
+                continue
+            drive_path = utils_misc.get_linux_drive_path(session, drive_id)
+            if not drive_path:
+                test.error("Failed to get '%s' drive path" % data_image)
+            yield drive_path[5:]
+
+    data_iamges = params["images"].split()[1:]
```
Comment (CongLi) [inline qemu/tests/fio_linux.py]: This is a typo, and others are same.
s/iamge/image

File: qemu/tests/cfg/fio_linux.cfg [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    images += " stg0"
+    image_name_stg0 = images/storage0
+    image_size_stg0 = 10G
+    blk_extra_params_stg0 = "serial=TARGET_DISK0"
+    remove_image_stg0 = yes
+    force_create_image_stg0 = yes
+    start_vm = yes
+    fio_default_options = '--direct=1 --bs=64K --size=1000M --name=test --iodepth=1 --ioengine=libaio'
```
Comment (CongLi) [inline qemu/tests/cfg/fio_linux.cfg]: Please update the size to 2G, iodepth=4

File: qemu/tests/fio_linux.py (hunk 3 above)
Comment (YongxueHong) [reply qemu/tests/fio_linux.py]: thanks , have updated.

File: qemu/tests/cfg/fio_linux.cfg (hunk 4 above)
Comment (YongxueHong) [reply qemu/tests/cfg/fio_linux.cfg]: thanks , have updated.

File: qemu/tests/cfg/fio_linux.cfg (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/cfg/fio_linux.cfg]: thanks , have updated.

**Comment by CongLi** [comment]
Ack.
Finshed=20, PASS=20

**Comment by ngu-niny** [comment]
Test Pass:
'1-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.fio_linux.aio_native' Finshed 2019-04-11 01:47:25 2019-04-11 01:57:16 PASS 591.829663
'2-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.fio_linux.aio_threads' Finshed 2019-04-11 01:57:16 2019-04-11 02:07:16 PASS 599.089081

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by XueqiangWei** [comment]
Tested on rhel7 with q35/pc + virtio-blk/virtio-scsi, all passed.

Ack

**Comment by YongxueHong** [comment]
thanks all.

**Pull request merged**